pkg/pf/tfbridge/provider_diff.go

```diff
@@ -208,3 +212,74 @@ func diffAttributePaths(tfDiff []tftypes.ValueDiff) []*tftypes.AttributePath {
 	}
 	return paths
 }
+
+func trimElementKeyValueFromTFPath(path *tftypes.AttributePath) *tftypes.AttributePath {
+	// trims everything after an ElementKeyValue path step from the replaceKeys paths
+	//nolint:lll
+	// see `convert.AttributePathToPath` used here: https://github.com/opentofu/opentofu/blob/5968e195b0f6e1ecb4381afa26abb95c636fe755/internal/plugin6/grpc_provider.go#L503
```

Comment:
This would need some more explanation I believe. `convert.AttributePathToPath` looks very different it seems. It does not do trimming or dropping steps.